DigitalLearningSolutions.Data.Tests/DataServices/LearningLogItemsDataServiceTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System;
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using NUnit.Framework;
+
+    public class LearningLogItemsDataServiceTests
+    {
+        private CompetencyLearningResourcesTestHelper competencyLearningResourcesTestHelper = null!;
+        private LearningLogItemsTestHelper learningLogItemsTestHelper = null!;
+        private ILearningLogItemsDataService service = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            var connection = ServiceTestHelper.GetDatabaseConnection();
+            service = new LearningLogItemsDataService(connection);
+
+            learningLogItemsTestHelper = new LearningLogItemsTestHelper(connection);
+            competencyLearningResourcesTestHelper = new CompetencyLearningResourcesTestHelper(connection);
+        }
+
+        [Test]
+        public void InsertLearningLogItem_inserts_expected_record()
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                // Given
+                const int competencyLearningResourceId = 1;
+                const int delegateId = 2;
+                const string resourceName = "Activity";
+                const string resourceLink = "www.test.com";
+                var addedDate = new DateTime(2021, 11, 1);
+                competencyLearningResourcesTestHelper.InsertCompetencyLearningResource(
+                    1,
+                    competencyLearningResourceId,
+                    1,
+                    7
+                );
+
+                // When
+                service.InsertLearningLogItem(
+                    delegateId,
+                    addedDate,
+                    resourceName,
+                    resourceLink,
+                    competencyLearningResourceId
+                );
+                var result = learningLogItemsTestHelper.SelectLearningLogItemWithResourceLink(resourceLink);
+
+                // Then
+                using (new AssertionScope())
+                {
+                    result.Should().NotBeNull();
+                    result!.LoggedById.Should().Be(delegateId);
+                    result.LoggedDate.Should().Be(addedDate);
+                    result.LinkedCompetencyLearningResourceId.Should().Be(competencyLearningResourceId);
+                    result.ExternalUri.Should().Be(resourceLink);
+                    result.Activity.Should().Be(resourceName);
+                    result.ActivityTypeId.Should().NotBe(0);

```

Comment:
I feel like the ActivityTypeId should be 8 explicitly, as it comes from a reference table, but I suppose we can't guarantee that given the migration. Do you think this is worth changing? An alternative might be making the query used JOIN onto the ActivityTypes table, and recovering the TypeLabel and making sure that is "Learning Hub Resource"

Reply:
Changed this to get the TypeLabel as I think it is safer than assuming an ID